src/products/dto/product.dto.ts

```diff
@@ -0,0 +1,16 @@
+import { IsNotEmpty, IsNumber, IsString } from 'class-validator';
+import { StatusEnum } from '../types/status-enum';
+
+export class CreateProductDto {
+  id: string;
+  @IsString()
+  @IsNotEmpty()
+  name: string;
```

Comment:
You can add description and example, because when you do the documentation in swagger, it will be correct.
```suggestion
  @IsString()
  @IsNotEmpty()
   @ApiProperty({ description: 'example description', example: 'foo' })
  name: string;
```